src/com/googlecode/jmxtrans/model/output/GangliaWriter.java

```diff
@@ -166,11 +169,14 @@ public void internalWrite(Server server, Query query, ImmutableList<Result> resu
 			if (result.getValues() != null) {
 				for (final Map.Entry<String, Object> resultValue : result.getValues().entrySet()) {
 					final String name = KeyUtils.getKeyString(query, result, resultValue, getTypeNames());
-					final String value = resultValue.getValue().toString();
+					
+					BigDecimal value = new BigDecimal(resultValue.getValue().toString());
+					String strValue = value.compareTo(GANGLIA_MIN_DOUBLE) <= 0 ? "0" : value.toString();
```

Comment:
Seems that when value is too small for Ganglia, you replace it with 0. What you probably want is to replace it with `GANGLIA_MIN_DOUBLE`. This should probably be implemented as `Math.max(resultValue.getValue(), GANGLIA_MIN_DOUBLE)`. Note that there is no garantee that `resultValue.getValue()` is a number, so some type checking should probably be done.